include/lsst/pex/exceptions/Runtime.h

```diff
@@ -32,17 +32,138 @@ namespace lsst {
 namespace pex {
 namespace exceptions {
 
+/**
+ * Reports errors in the logical structure of the program.
+ *
+ * LogicError and its subclasses should be thrown to represent problems, such
+ * as violation of logical preconditions or class invariants, that are in
+ * principle preventable using defensive programming or other good practices.
+ * In many cases, it may not be appropriate to catch them.
+ *
+ * @see RuntimeError
+ * @see std::logic_error
```

Comment:
Class relationships are prominently displayed in the documentation, so it will be clear that it's not (nor would that make sense?). But I felt it necessary to indicate somehow that most of these exceptions are C++-inspired, especially for readers not familiar with the standard C++ exceptions.